docs/docs/client-only-routes-and-user-authentication.md

```diff
@@ -2,25 +2,121 @@
 title: Client-only Routes & User Authentication
 ---
 
-Often you want to create a site with client-only portions that are gated by authentication.
+Often you want to create a site with client-only portions, which allows you to gate them by authentication or load different content based on URL parameters.
+
+## Understanding client-only routes
 
 A classic example would be a site that has a landing page, various marketing pages, a login page, and then an app section for logged-in users. The logged-in section doesn't need to be server rendered as all data will be loaded live from your API after the user logs in. So it makes sense to make this portion of your site client-only.
 
-Gatsby uses [@reach/router](https://reach.tech/router/) under the hood. You should use @reach/router to create client-only routes.
+Client-only routes will exist on the client only and will not correspond to index.html files in an app's built assets. If you'd like site users to be able to visit client routes directly, you need to set up your site to handle those routes appropriately.
+
+A sample site might be set up like this:
+
+![Site with a static homepage and client-only routes](./images/client-only-routes.png)
+
+Gatsby converts components in the `pages` folder into static HTML files for the Home page and the App page. A `<Router />` is added to the App page so that the profile and details components can be rendered from the App page, they don't have static assets built for them because they exist only on the client. The profile page can `POST` data about a user back to an API, and the details page can dynamically load data about a user with a specific id from an API.
```

Comment:
```suggestion
Gatsby converts components in the `pages` folder into static HTML files for the Home page and the App page. A `<Router />` is added to the App page so that the profile and details components can be rendered from the App page; they don't have static assets built for them because they exist only on the client. The profile page can `POST` data about a user back to an API, and the details page can dynamically load data about a user with a specific id from an API.
```